www/host.js

```diff
@@ -56,22 +57,31 @@ socket.onmessage = (event) => {
     // TODO: Do some validation on the payload data I guess.
     let payload = JSON.parse(event.data);
 
-    if (payload['PlayerRegistered']) {
-        registerPlayer(payload['PlayerRegistered']);
-    } else if (payload['PlayerScore']) {
-        let info = payload['PlayerScore'];
+    if (payload['PlayerRegister']) {
+        registerPlayer(payload['PlayerRegister']);
+    } else if (payload['AddBall']) {
+        let info = payload['AddBall'];
+
+        // Find the hippo/player for the player that scored.
+        let hippo = app.hippoMap[info.id];
+        assert(hippo != null, 'Unable to find hippo for ID: ' + info.id);
+
+        hippo.player.balls = info.balls;
+    } else if (payload['HippoEat']) {
+        let info = payload['HippoEat'];
 
         // Find the hippo/player for the player that scored.
         let hippo = app.hippoMap[info.id];
         assert(hippo != null, 'Unable to find hippo for ID: ' + info.id);
 
         // Updated the local score for the player.
         hippo.player.score = info.score;
+        hippo.player.balls = info.balls;
 
         // Animate the hippo head to match the score increase. The direction of the chomp animation
         // depends on the side of the screen that the hippo is on.
         let element = document.getElementById(hippo.player.id);
-        switch (hippo.side) {
+        switch (hippo.side.side) {
```

Comment:
`hippo.side` is an object: `{ side: number, array: Hippo[] }`. That could be renamed to `hippo.side.sideId` or something to make it clearer.